### ISSUE TITLE
各种主干网络是否有预训练权重应该在哪看

### ISSUE THREAD

**Issue by toouki** [PaddlePaddle/PaddleSeg#3138]
### 问题确认 Search before asking

- [X] 我已经搜索过问题，但是没有找到解答。I have searched the question and found no related answer.

### 请提出你的问题 Please ask your question

请问各种主干网络是否有预训练权重应该在哪看？
我只能从config中翻到一些

**Comment by marshall-dteach** [comment]
你好，我来回答一下，各个主干网络的与训练权重大部分都是从现有torch模型保存的模型转化来的，如果有需要可以下载对应的pth模型通过代码转化，我这边提供一个简单版本的模型转化，供你参考（实际过程需要你根据模型的信息调整一些内容）。

**Comment by marshall-dteach** [comment]
你好，我来回答一下，各个主干网络的与训练权重大部分都是从现有torch模型保存的模型转化来的，如果有需要可以下载对应的pth模型通过代码转化，我这边提供一个简单版本的模型转化，供你参考（实际过程需要你根据模型的信息调整一些内容）。

**Comment by marshall-dteach** [comment]
你好，我来回答一下，各个主干网络的与训练权重大部分都是从现有torch模型保存的模型转化来的，如果有需要可以下载对应的pth模型通过代码转化，我这边提供一个简单版本的模型转化，供你参考（实际过程需要你根据模型的信息调整一些内容）。
```python```
##pth -> pdparams
import torch
import numpy as np
import paddle

model_dict = torch.load('pth/eformer_s0_450.pth')['model']
paddle_dict = {}
for key, value in model_dict.items():
    if "running_mean" in key:
        key = key.replace("running_mean", "_mean")
    elif "running_var" in key:
        key = key.replace("running_var", "_variance")
    # if len(value.shape) == 2:
    #     paddle_dict[key] = np.transpose(value.detach().cpu().numpy(), ([1, 0]))
    if len(value.shape) == 0:
        print('jump',value, key)
        continue
    else:
        paddle_dict[key] = value.detach().cpu().numpy()
    # print(key, value.shape)
print(len(paddle_dict.keys()))
paddle.save(paddle_dict, "pdparams/eformer_s0_450.pdparams")
```
对应的模型是(efficientformerV2)[https://github.com/snap-research/EfficientFormer]的模型，可从此处下载
![image](https://github.com/PaddlePaddle/PaddleSeg/assets/63546191/a07700c3-cd72-4799-9108-07dd1da024b3)

**Comment by marshall-dteach** [comment]
你好，我来回答一下，各个主干网络的与训练权重大部分都是从现有torch模型保存的模型转化来的，如果有需要可以下载对应的pth模型通过代码转化，我这边提供一个简单版本的模型转化，供你参考（实际过程需要你根据模型的信息调整一些内容）。
```python```
##pth -> pdparams
import torch
import numpy as np
import paddle

model_dict = torch.load('pth/eformer_s0_450.pth')['model']
paddle_dict = {}
for key, value in model_dict.items():
if "running_mean" in key:
key = key.replace("running_mean", "_mean")
elif "running_var" in key:
key = key.replace("running_var", "_variance")
# if len(value.shape) == 2:
# paddle_dict[key] = np.transpose(value.detach().cpu().numpy(), ([1, 0]))
if len(value.shape) == 0:
print('jump',value, key)
continue
else:
paddle_dict[key] = value.detach().cpu().numpy()
# print(key, value.shape)
print(len(paddle_dict.keys()))
paddle.save(paddle_dict, "pdparams/eformer_s0_450.pdparams")
```

**Comment by marshall-dteach** [comment]
&emsp;&emsp;你好，我来回答一下，各个主干网络的与训练权重大部分都是从现有torch模型保存的模型转化来的，如果有需要可以下载对应的pth模型通过代码转化，我这边提供一个简单版本的模型转化，供你参考（实际过程需要你根据模型的信息调整一些内容）。
```python
##pth -> pdparams
import torch
import numpy as np
import paddle

model_dict = torch.load('pth/eformer_s0_450.pth')['model']
paddle_dict = {}
for key, value in model_dict.items():
    if "running_mean" in key:
        key = key.replace("running_mean", "_mean")
    elif "running_var" in key:
    key = key.replace("running_var", "_variance")
    # if len(value.shape) == 2:
        # paddle_dict[key] = np.transpose(value.detach().cpu().numpy(), ([1, 0]))
    if len(value.shape) == 0:
        print('jump',value, key)
        continue
    else:
        paddle_dict[key] = value.detach().cpu().numpy()
# print(key, value.shape)
print(len(paddle_dict.keys()))
paddle.save(paddle_dict, "pdparams/eformer_s0_450.pdparams")
```

**Comment by marshall-dteach** [comment]
&emsp;&emsp;你好，我来回答一下，各个主干网络的与训练权重大部分都是从现有torch模型保存的模型转化来的，如果有需要可以下载对应的pth模型通过代码转化，我这边提供一个简单版本的模型转化，供你参考（实际过程需要你根据模型的信息调整一些内容）。
```python
##pth -> pdparams
import torch
import numpy as np
import paddle

model_dict = torch.load('pth/eformer_s0_450.pth')['model']
paddle_dict = {}
for key, value in model_dict.items():
    if "running_mean" in key:
        key = key.replace("running_mean", "_mean")
    elif "running_var" in key:
    key = key.replace("running_var", "_variance")
    # if len(value.shape) == 2:
        # paddle_dict[key] = np.transpose(value.detach().cpu().numpy(), ([1, 0]))
    if len(value.shape) == 0:
        print('jump',value, key)
        continue
    else:
        paddle_dict[key] = value.detach().cpu().numpy()
# print(key, value.shape)
print(len(paddle_dict.keys()))
paddle.save(paddle_dict, "pdparams/eformer_s0_450.pdparams")
```
&emsp;$emsp;对应的模型项目链接为：[efficientformerV2](https://github.com/snap-research/EfficientFormer)，可以从图中位置下载模型<br>
![image](https://github.com/PaddlePaddle/PaddleSeg/assets/63546191/253c962a-552d-401a-ac8b-0b497d339bbf)

**Comment by shiyutang** [comment]
以上回答已经充分解答了问题，如果有新的问题欢迎随时提交issue，或者在此条issue下继续回复～

**Comment by shiyutang** [comment]
我们开启了飞桨套件的ISSUE攻关活动，欢迎感兴趣的开发者参加：https://github.com/PaddlePaddle/PaddleOCR/issues/10223
